### PR TITLE
OAuth2Implementation

### PR DESCRIPTION
**Pull request by manivinesh**

### CODE REVIEW

File: lib/ET_Client.js [hunk 1]
```diff
@@ -208,8 +216,7 @@ ET_Client.prototype.initSoap = function(auth, origin, stack, soapOrigin, globalR
 		headers: globalReqOptions.headers,
 		globalReqOptions
     });
-    client.origin = origin;
-
+    client.origin = soapOrigin;
```
Comment (sfdrogojan) [inline lib/ET_Client.js]: Why is this needed?

File: lib/ET_Client.js [hunk 2]
```diff
@@ -142,12 +142,15 @@ var _id='';
 var _secret='';
 var _stack='';
 
-var ET_Client = function(clientId, clientSecret, stack, {origin, authOrigin, soapOrigin, globalReqOptions = {}} = {}) {
+var ET_Client = function(clientId, clientSecret, stack, {origin, authOrigin, soapOrigin, authOptions = {} = {}, globalReqOptions = {}} = {}) {
```
[the remark this answers is not in the document]
Comment (manivinesh) [reply lib/ET_Client.js]: yeah, thats s mistake. will remove it.

File: test.xml [hunk 3]
```diff
@@ -0,0 +1 @@
+<Envelope xmlns="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">\n  <Body>\n    <RetrieveRequestMsg xmlns="http://exacttarget.com/wsdl/partnerAPI">\n      <RetrieveRequest>\n        <ObjectType>BounceEvent</ObjectType>\n        <Properties>ID</Properties>\n      </RetrieveRequest>\n    </RetrieveRequestMsg>\n  </Body>\n  <Header>\n    <fueloauth xmlns="http://exacttarget.com">eyJhbGciOiJIUzI1NiIsImtpZCI6IjIiLCJ2ZXIiOiIxIiwidHlwIjoiSldUIn0.eyJhY2Nlc3NfdG9rZW4iOiJhbFVzR0VtZjl5a3A4WFp2MjF6NjE0OGgiLCJjbGllbnRfaWQiOiI1ODI0ZjR0bTF6ZG9hYjZibGt0N2ZycDkiLCJlaWQiOjEwNzcyNTk0LCJzdGFja19rZXkiOiJRQTFTMSIsInBsYXRmb3JtX3ZlcnNpb24iOjIsImNsaWVudF90eXBlIjoiU2VydmVyVG9TZXJ2ZXIifQ.__a84C03A43NhoN_M58zzJHUDs_1VuMtYk0wM2bmesg.pKA3lWUOUxdJpLXJICIgjFIt2wlfWHF5mHuli7ZlXh_Apn7oahVFwthaL73W_y_Nlf_6C9HZv0svvv3qacdg40aWCHqkOCE85xgwBCmzo5hM2Pa2a1JN5uYyuKwYW1kR-kt2QPa0KV3dviO5qboifBpknZHRqXj0SCGTcX91JIy_HwrdP</fueloauth>\n  </Header>\n</Envelope>
```
Comment (sfdrogojan) [inline test.xml]: Is this file used somewhere?

File: test/.npmrc [hunk 4]
```diff
@@ -0,0 +1,4 @@
+_auth=${ARTIFACTORY_AUTH}
```
Comment (sfdrogojan) [inline test/.npmrc]: Is this file needed?

File: test.xml (hunk 3 above)
Comment (manivinesh) [reply test.xml]: No. I'll remove it

**Comment by manivinesh** [comment]
@sfdrogojan i think i covered all code review concerns
